feat(weave): Video display in weave

zbirenbaum opened the pull request:
## Description

Adds a video panel to trace view and a link which opens a modal to the trace table cells containing a video ref

<!--
Include reference to internal ticket "Fixes WB-NNNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
-->

- Fixes WB-24814

zbirenbaum commented:
Added Views:
Trace Table
![image](https://github.com/user-attachments/assets/d31ca22f-9ba6-4cc5-8ea7-29795e836576)

On-click `MP4 Video` link in trace table
![image](https://github.com/user-attachments/assets/1d3f1b79-7725-4938-bbbb-b90c22220a1e)

Trace View
![image](https://github.com/user-attachments/assets/a8246fec-8f7a-46cb-b137-2bcc70f31823)

neutralino1 commented on weave-js/src/components/PagePanelComponents/Home/Browse3/typeViews/moviepy.video.VideoClip/VideoPlayer.tsx:
```diff
@@ -0,0 +1,188 @@
+    return (
+      <>
+        <div
+          style={{
+            display: 'flex',
+            justifyContent: 'flex-start',
+            width: '100%',
+            margin: 'auto',
+            padding: '6px',
```
You use tailwind below, can you use it here as well?

neutralino1 commented on weave-js/src/components/PagePanelComponents/Home/Browse3/typeViews/moviepy.video.VideoClip/VideoPlayer.tsx:
```diff
@@ -0,0 +1,188 @@
+            margin: 'auto',
+            padding: '6px',
+          }}>
+          <CustomLink text={videoText} onClick={() => setShowPopup(true)} />
```
Any chance we could make the link NOT bold? To differentiate from the call name link at the beginning of the row?

neutralino1 commented on weave-js/src/components/PagePanelComponents/Home/Browse3/typeViews/moviepy.video.VideoClip/VideoPlayer.tsx:
```diff
@@ -0,0 +1,188 @@
+  const title = data.custom_name || entity.split('-')[0];
+
+  // Link mode (default view)
+  if (!mode || mode !== 'object_viewer') {
```
I wonder if it would be wiser to have two components, one for link view and one for player view, instead of one with a `if`.
It seems the link needs much less code, so most of this component would be unused, which could become heavy in a long list of calls.

zbirenbaum replied:
That would be ideal. Unfortunately it's a bit difficult. The code is set up s.t. there's a single component which is used to render the CustomWeaveType. The only way we really have of controlling how this renders is by checking the `mode`.

This context already existed even though most components don't use it, so someone has definitely run into this issue before...

neutralino1 reviewed (changes requested):
A couple of notes.

zbirenbaum commented:
I addressed the comments with the exception of the separate components one since I'm not sure that can be easily done. I also added a play button but I'm not sure how I feel about it...
![image](https://github.com/user-attachments/assets/8fc3e15f-12e6-47ec-a8f3-f1a00c449ecc)
